instance/management/commands/update_metadata.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2018 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance app - instances' metadata update management command
+"""
+
+# Imports #####################################################################
+
+import sys
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from instance.models.openedx_instance import OpenEdXInstance
+
+
+# Classes #####################################################################
+
+
+class Command(BaseCommand):
+    """
+    management command class for updating instances metadata
+    """
+    help = 'Updates the Consul metadata for all instances or just a single instance.'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-i',
+            '--instance',
+            type=int,
+            required=False,
+            help='Specific Instance ID you want to update its metadata.'
+        )
+        parser.add_argument(
+            '--skip-clean',
+            action='store_true',
+            help='Will skip cleaning metadata for archived instances.'
+        )
+        parser.add_argument(
+            '--skip-update',
+            action='store_true',
+            help='Will skip cleaning metadata for archived instances.'
+        )
+
+    def handle(self, *args, **options):
+        if not settings.CONSUL_ENABLED:
+            self.stdout.write(self.style.WARNING('Consider Enable Consul before using this command.'))
+            sys.exit(0)
+
+        instance_id = options.get('instance')
+
+        if not options.get('skip_update'):
+            if instance_id:
+                self.update_instance_metadata(instance_id)
+            else:
+                self.update_all_instances_metadata()
+
+        if not options.get('skip_clean'):
```

Comment:
I think it's somewhat unexpected that old data of unrelated instances will be cleaned when specifying an instance id on the command line.

(There isn't a strong need for all the command line flags anyway – in general, we always want all metadata to be up to date.  So it's a bit hard to argue about the semantics without having a use case in mind.)

Reply:
You are right, this is why I left it like this without adding a check for the instance flag, and I also thought that having Consul clean is good in all cases. However, it's really confusing to see that you're touching multiple instances if you specified a single instance to work with. I'll update it now.

Reply:
I removed the support for a single instance in this command, because a user who's updating the metadata for an instance doesn't expect the data to go through the purge functionality and vice versa. Until we have a real need for it with a clear requirement, we can re-do it again.